projects/observability/src/shared/components/explore-query-editor/order-by/explore-query-order-by-editor.component.ts

```diff
@@ -0,0 +1,239 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+import { TypedSimpleChanges } from '@hypertrace/common';
+import { SelectOption } from '@hypertrace/components';
+import { combineLatest, EMPTY, Observable, of, ReplaySubject, Subject } from 'rxjs';
+import { map, switchMap, take } from 'rxjs/operators';
+import { AttributeMetadata } from '../../../graphql/model/metadata/attribute-metadata';
+import { getAggregationDisplayName, MetricAggregationType } from '../../../graphql/model/metrics/metric-aggregation';
+import { GraphQlSortDirection } from '../../../graphql/model/schema/sort/graphql-sort-direction';
+import { TraceType } from '../../../graphql/model/schema/trace';
+import { MetadataService } from '../../../services/metadata/metadata.service';
+import { GraphQlOrderBy } from '../explore-visualization-builder';
+@Component({
+  selector: 'ht-explore-query-order-by-editor',
+  styleUrls: ['./explore-query-order-by-editor.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="order-by-container" *ngIf="this.orderByExpression">
+      <span class="order-by-main-label"> Order by </span>
+      <div class="fields-container">
+        <div class="order-by-input-container" *ngIf="this.metricOptions$ | async as metricOptions">
+          <span class="order-by-label"> Metric </span>
+          <ht-select
+            class="attribute-selector"
+            [selected]="this.selectedMetric$ | async"
+            [showBorder]="true"
+            (selectedChange)="this.onMetricChange($event)"
+          >
+            <ht-select-option
+              *ngFor="let metricOption of metricOptions"
+              [value]="metricOption.value"
+              [label]="metricOption.label"
+            ></ht-select-option>
+          </ht-select>
+        </div>
+        <div class="order-by-input-container" *ngIf="this.aggregationOptions$ | async as aggregationOptions">
+          <span class="order-by-label"> Aggregation </span>
+          <ht-select
+            class="aggregation-selector"
+            [selected]="this.selectedAggregation$ | async"
+            [showBorder]="true"
+            (selectedChange)="this.onAggregationChange($event)"
+            [disabled]="aggregationOptions && aggregationOptions.length === 1"
+          >
+            <ht-select-option
+              *ngFor="let aggregationOption of aggregationOptions"
+              [value]="aggregationOption.value"
+              [label]="aggregationOption.label"
+            ></ht-select-option>
+          </ht-select>
+        </div>
+        <div class="order-by-input-container" *ngIf="this.sortByOptions$ | async as sortByOptions">
```

Comment:
these don't really need to come from an observable, there's always going to be exactly two - ascending and descending.

Reply:
there are three options indeed: None, asc, desc

Reply:
done